src/main/java/com/zaxxer/hikari/pool/HikariPool.java

```diff
@@ -567,13 +559,15 @@ protected void update() {
       @Override
       public Boolean call() throws Exception
       {
+         // Speculative increment of totalConnections with expectation of success
+         if (totalConnections.incrementAndGet() > config.getMaximumPoolSize()) {
+            totalConnections.decrementAndGet(); // Pool is at max size
+            return Boolean.FALSE;
+         }
```

Comment:
Exactly what I said before 'total connections reported is not 'actual', it includes sleeping add requests'
needs some more workout to iron out even this flaw, your input is requested :)